### PR TITLE
Java11

### PR DESCRIPTION
**Pull request by sbusk**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -75,6 +76,8 @@
 
         <mail.version>1.6.2</mail.version>
         <mariadb-java-client.version>2.5.1</mariadb-java-client.version>
+
+        <!-- TODO: [ES] update mockito to 2.21.0 (need to fix mocks in tests) -->
```
Comment (eshryane) [inline pom.xml:80]: FYI there is a mockito2 branch (in progress) for this

File: pom.xml [hunk 2]
```diff
@@ -818,6 +827,7 @@
                     </excludes>
                     <runOrder>random</runOrder>
                     <threadCount>1</threadCount>
+                    <forkCount>0</forkCount>
```
Comment (eshryane) [inline pom.xml:830]: Why no fork?

File: pom.xml [hunk 3]
```diff
@@ -828,14 +838,21 @@
...
+                </dependencies>
             </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-failsafe-plugin</artifactId>
                 <version>2.22.2</version>
                 <configuration>
                     <redirectTestOutputToFile>${test.redirectOutput}</redirectTestOutputToFile>
-                    <argLine>-Xmx1024m -DjvmId=${surefire.forkNumber} ${failsafe.jacoco.args}</argLine>
+                    <argLine>-Xmx1024m -DjvmId=${surefire.forkNumber} --add-modules java.sql,java.xml ${failsafe.jacoco.args}</argLine>
```
Comment (eshryane) [inline pom.xml:855]: These arguments are commented out in the build step above (--add-modules java.sql,java.xml ), are they needed here?

Comment (sbusk) [reply pom.xml:855]: They will be when we start compiling for Java11.

File: pom.xml (hunk 2 above)
Comment (sbusk) [reply pom.xml:830]: Idk, didn't touch this part of the code/build..

Comment (sbusk) [reply pom.xml:830]: Merge error maybe?

Comment (eshryane) [reply pom.xml:830]: Hmm I can't remember what I did there, can you use the config from master and see if that's working?

Comment (eshryane) [reply pom.xml:830]: I made the change in bf54d71

File: pom.xml [hunk 4]
```diff
@@ -25,11 +25,12 @@
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
 
         <!-- JAVA version -->
-        <java.version>1.8</java.version>
+        <java.version>11</java.version>
 
         <!-- dependencies -->
 
-        <aspectj.version>1.8.8</aspectj.version>
+        <aspectj.version>1.9.4</aspectj.version>
```
Comment (eshryane) [inline pom.xml:32]: Latest AspectJ is now 1.9.6 (according to dependabot)

**Comment by eshryane** [comment]
Test failure:
`2094[ERROR] testChannelHandlersAddedToPipeline(net.ripe.db.whois.nrtm.BaseNrtmServerPipelineFactoryIntegrationTest)  Time elapsed: 0.435 s  <<< FAILURE!
2095java.lang.AssertionError: Field 'ajc$tjp_0' in channel handler net.ripe.db.whois.nrtm.NrtmQueryHandler must be volatile or final
2096	at net.ripe.db.whois.nrtm.BaseNrtmServerPipelineFactoryIntegrationTest.testChannelHandlersAddedToPipeline(BaseNrtmServerPipelineFactoryIntegrationTest.java:48)
`

Also happening on dependabot/maven/aspectj.version-1.9.6 branch (see #676).

File: pom.xml (hunk 2 above)
Comment (sbusk) [reply pom.xml:837]: The forkCount does not exist on master..

File: pom.xml (hunk 4 above)
Comment (sbusk) [reply pom.xml:32]: Bumped, thanks

Comment (eshryane) [reply pom.xml:32]: I merged #676 the latest is 1.9.6.

File: pom.xml (hunk 2 above)
Comment (eshryane) [reply pom.xml:837]: Good point! Better to leave it out, than disable it using zero.

**Comment by sbusk** [comment]
> Test failure:
> `2094[ERROR] testChannelHandlersAddedToPipeline(net.ripe.db.whois.nrtm.BaseNrtmServerPipelineFactoryIntegrationTest) Time elapsed: 0.435 s <<< FAILURE! 2095java.lang.AssertionError: Field 'ajc$tjp_0' in channel handler net.ripe.db.whois.nrtm.NrtmQueryHandler must be volatile or final 2096 at net.ripe.db.whois.nrtm.BaseNrtmServerPipelineFactoryIntegrationTest.testChannelHandlersAddedToPipeline(BaseNrtmServerPipelineFactoryIntegrationTest.java:48) `
> 
> Also happening on dependabot/maven/aspectj.version-1.9.6 branch (with potential fix) (see #676).
> 
> This also happens locally in IntelliJ (make sure to select the ajc compiler, point to the aspectjtools 1.9.6, and run all the whois-nrtm tests at once).

I managed to reproduce it, thanks.

**Review by eshryane — APPROVED** [review]

**Comment by eshryane** [comment]
Added dependency on asm in maven-compile-plugin and maven-surefire-plugin:

"maven-compiler-plugin depends on the old version of ASM which does not support Java 10 (and Java 11) yet. However, it is possible to explicitly specify the right version of ASM..."

https://stackoverflow.com/questions/49398894/unable-to-compile-simple-java-10-java-11-project-with-maven

This explicit dependency is no longer needed.

**Pull request merged**